cpp/src/rolling/detail/optimized_unbounded_window.hpp

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/column/column.hpp>
+#include <cudf/types.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+
+namespace rmm::mr {
+class device_memory_resource;
+}
+
+namespace cudf {
+
+class rolling_aggregation;
+class table_view;
+struct window_bounds;
```

Comment:
Can this be deleted? Or do consumers of these APIs need this struct to exist for some reason?
```suggestion
```

Reply:
Good catch. This is an artifact from a previous refactor.

Reply:
Removed.